tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -319,4 +319,89 @@ public function testListFloatTypeColumns()
+        $this->assertNull($onlineTable->getColumn('col5')->getDefault());
+        $this->assertEquals(-2.3, $onlineTable->getColumn('col6')->getDefault());
+        $this->assertEquals('2012-12-12', $onlineTable->getColumn('col7')->getDefault());
+        $this->assertTrue($onlineTable->getColumn('col8')->getNotnull());
+        $this->assertEquals('', $onlineTable->getColumn('col8')->getDefault());
+        $this->assertEquals(0, $onlineTable->getColumn('col9')->getDefault());
+        $this->assertEquals('He"ll"o world', $onlineTable->getColumn('col10')->getDefault());
+
+        $comparator = new Comparator();
+
+        $diff = $comparator->diffTable($table, $onlineTable);
+        $this->assertFalse($diff, "Tables should be identical with column defaults.");
+    }
+
+    /**
+     * For testing MariaDB 10.2+ / MySQL differences
+     */
+    public function testColumnDefaultsWithCreate() {
+
+        $this->_conn->query('DROP TABLE IF EXISTS `test_column_defaults_with_create`');
+        $sql = sprintf("
+            CREATE TABLE `test_column_defaults_with_create` (
+                `col0` INT(11) NULL,
+                `col1` INT(10) UNSIGNED NULL DEFAULT NULL,
+                `col2` VARCHAR(10) CHARACTER SET UTF8MB4 COLLATE UTF8MB4_GENERAL_CI NULL DEFAULT NULL,
+                `col3` VARCHAR(10) CHARACTER SET UTF8MB4 COLLATE UTF8MB4_GENERAL_CI NULL DEFAULT 'NULL',
+                `col4` VARCHAR(20) NULL DEFAULT 'Hello world',
+                `col5` DATETIME NULL DEFAULT NULL,
+                `col6` DECIMAL(6, 3) NULL DEFAULT -2.3,
+                `col7` DATE NULL DEFAULT '2012-12-12',
+                `col8` VARCHAR(20) NOT NULL DEFAULT '',
+                `col9` INT(11) NULL DEFAULT 0,
+                `col10` VARCHAR(20) NULL DEFAULT '%s'                                                
+                ) ENGINE=INNODB;
+        ", 'He"ll"o world');
+
+        $this->_conn->query($sql);
+        $t = $this->_sm->listTableDetails("test_column_defaults_with_create");
+        $this->assertNull($t->getColumn('col0')->getDefault());
+        $this->assertNull($t->getColumn('col1')->getDefault());
+        $this->assertNull($t->getColumn('col2')->getDefault());
+        $this->assertEquals('NULL', $t->getColumn('col3')->getDefault());
+        $this->assertEquals('Hello world', $t->getColumn('col4')->getDefault());
+        $this->assertNull($t->getColumn('col5')->getDefault());
+        $this->assertEquals(-2.3, $t->getColumn('col6')->getDefault());
+        $this->assertEquals('2012-12-12', $t->getColumn('col7')->getDefault());
+        $this->assertTrue($t->getColumn('col8')->getNotnull());
+        $this->assertEquals('', $t->getColumn('col8')->getDefault());
+        $this->assertEquals(0, $t->getColumn('col9')->getDefault());
+        $this->assertEquals('He"ll"o world', $t->getColumn('col10')->getDefault());
+
+        // @todo mariadb returns 'current_timestamp()'
```

Comment:
What about this? Blocker for merging?

Reply:
mmm, let me one day more to see what I can do... From what I've seen it's not easy... The two syntaxes are accepted by mysql/mariadb. The only change is that with Mariadb, whatever syntax you used to create the table, it will always return `current_timestamp()` (don't remember where I read it, but from what I tested it's working like that).